### PR TITLE
[Teams] Add update and delete bot scenario

### PR DESCRIPTION
**Pull request by EricDahlvang**
Fixes https://github.com/microsoft/botbuilder-js/issues/1199

### CODE REVIEW

File: libraries/botbuilder-core/src/turnContext.ts [hunk 1]
```diff
@@ -320,7 +320,11 @@ export class TurnContext {
      * @param activity New replacement activity. The activity should already have it's ID information populated.
      */
     public updateActivity(activity: Partial<Activity>): Promise<void> {
-        return this.emit(this._onUpdateActivity, activity, () => this.adapter.updateActivity(this, activity));
+
+        let conversationReference = TurnContext.getConversationReference(this._activity);
```
Comment (stevengum) [inline libraries/botbuilder-core/src/turnContext.ts]: @Virtual-Josh and @EricDahlvang this change is the same as https://github.com/microsoft/botbuilder-js/pull/1169, correct? 

... It's a race! 

@EricDahlvang can you change the `let` to `const` in this method?

Comment (EricDahlvang) [reply libraries/botbuilder-core/src/turnContext.ts]: We took #1169 ... rebasing this PR with those changes (Kyle's implementation is cleaner :) )

File: libraries/botbuilder/tests/teams/activityUpdateAndDelete/src/activityUpdateAndDeleteBot.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            if (context.activity.text == "delete")
+            {
+                for (const activityId of this.activityIds) {
+                    await context.deleteActivity(activityId);
+                }
+
+                this.activityIds = [];
+            }
+            else
+            {
```
Comment (stevengum) [inline libraries/botbuilder/tests/teams/activityUpdateAndDelete/src/activityUpdateAndDeleteBot.ts]: These curly brackets 😭

File: libraries/botbuilder/tests/teams/activityUpdateAndDelete/src/activityUpdateAndDeleteBot.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+                await this.sendMessageAndLogActivityId(context, `${context.activity.text}`);
+
+                for (const activityId of this.activityIds) {
+                    let newActivity = MessageFactory.text(context.activity.text);
+                    newActivity.id = activityId;
+                    await context.updateActivity(newActivity);
+                }
```
Comment (stevengum) [inline libraries/botbuilder/tests/teams/activityUpdateAndDelete/src/activityUpdateAndDeleteBot.ts]: You could do the following...

```ts
const text = context.activity.text;
for (const id of this.activityIds) {
    await context.updateActivity({ id, text });
}
```

At the very least, please get rid of the MessageFactory, it really doesn't add anything here...

File: libraries/botbuilder/tests/teams/activityUpdateAndDelete/src/activityUpdateAndDeleteBot.ts [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+            }
+
+            // By calling next() you ensure that the next BotHandler is run.
+            await next();
+        });
+    }
+
+    async sendMessageAndLogActivityId(context: TurnContext, text: string): Promise<void> {
+        var replyActivity = MessageFactory.text(text);
+        var resourceResponse = await context.sendActivity(replyActivity);
```
Comment (stevengum) [inline libraries/botbuilder/tests/teams/activityUpdateAndDelete/src/activityUpdateAndDeleteBot.ts]: `const resourceResponse = await context.sendActivity({ text });`

Or just `const resourceResponse = await context.sendActivity(text);`

Go JavaScript (and overloads)!

File: libraries/botbuilder/tests/teams/activityUpdateAndDelete/src/activityUpdateAndDeleteBot.ts [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+        await this.activityIds.push(resourceResponse.id);
+
+        return;
```
Comment (stevengum) [inline libraries/botbuilder/tests/teams/activityUpdateAndDelete/src/activityUpdateAndDeleteBot.ts]: You can delete these two lines, you don't need to call `return;` 😄

Comment (EricDahlvang) [reply libraries/botbuilder/tests/teams/activityUpdateAndDelete/src/activityUpdateAndDeleteBot.ts]: removed return

File: libraries/botbuilder/tests/teams/activityUpdateAndDelete/src/activityUpdateAndDeleteBot.ts (hunk 4 above)
Comment (EricDahlvang) [reply libraries/botbuilder/tests/teams/activityUpdateAndDelete/src/activityUpdateAndDeleteBot.ts]: changed to use overload

File: libraries/botbuilder/tests/teams/activityUpdateAndDelete/src/activityUpdateAndDeleteBot.ts (hunk 3 above)
Comment (EricDahlvang) [reply libraries/botbuilder/tests/teams/activityUpdateAndDelete/src/activityUpdateAndDeleteBot.ts]: removed MessageFactory

File: libraries/botbuilder/tests/teams/activityUpdateAndDelete/src/activityUpdateAndDeleteBot.ts (hunk 2 above)
Comment (EricDahlvang) [reply libraries/botbuilder/tests/teams/activityUpdateAndDelete/src/activityUpdateAndDeleteBot.ts]: fixed curly brackets

**Pull request merged**